### PR TITLE
New UI: Run analyzers from the UI

### PR DESCRIPTION
**Pull request by berggren**
This PR enables the user to manually start analyzers on timelines within a sketch. It also adds a view to see the history of all other analyzers for that timeline.

Screenshot 2019-10-31 at 23 15 53

fixes: #997

### CODE REVIEW

File: timesketch/api/v1/resources.py [hunk 1]
```diff
@@ -1753,8 +1782,51 @@ def post(self):
...
+    @login_required
+    def get(self, sketch_id, timeline_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            An analysis in JSON (instance of flask.wrappers.Response)
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        timeline = Timeline.query.get(timeline_id)
+        analyses = Analysis.query.filter_by(timeline=timeline).all()
```
Comment (kiddinn) [inline timesketch/api/v1/resources.py]: s/analyses/ ... to something else... analysis_status ? or something like that?

File: timesketch/api/v1/resources.py [hunk 2]
```diff
@@ -1753,8 +1782,51 @@ def post(self):
...
+
+        if not sketch.has_permission(current_user, 'read'):
```
Comment (kiddinn) [inline timesketch/api/v1/resources.py]: can you move this above .... you've got

sketch = ....
if not sketch....

that is to abort early, before you start doing any queries, if you know the user doesn't even have the aCL to perform the action

File: timesketch/api/v1/resources.py [hunk 3]
```diff
@@ -1753,8 +1782,51 @@ def post(self):
...
+    def get(self, sketch_id, session_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            A analyzer session in JSON (instance of flask.wrappers.Response)
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        session = AnalysisSession.query.get(session_id)
+
+        if not sketch.has_permission(current_user, 'read'):
```
Comment (kiddinn) [inline timesketch/api/v1/resources.py]: same ehre.. .move thios check above the "session = ..."

**Comment by kiddinn** [comment]
@berggren also take a look at the travis run

File: timesketch/api/v1/resources.py (hunk 1 above)
Comment (berggren) [reply timesketch/api/v1/resources.py]: Done

File: timesketch/api/v1/resources.py (hunk 2 above)
Comment (berggren) [reply timesketch/api/v1/resources.py]: Done

File: timesketch/api/v1/resources.py (hunk 3 above)
Comment (berggren) [reply timesketch/api/v1/resources.py]: Done

**Comment by berggren** [comment]
@kiddinn PTAL

**Pull request merged**